backend/schedule/models.py

```diff
@@ -131,6 +131,13 @@ class ScheduleItem(TimeStampedModel):
         on_delete=models.PROTECT,
     )
 
+    subscribed_users = models.ManyToManyField(
```

Comment:
quick look and a nitpick: I am not sure we should save the preferences on the schedule item, I would keep the user part and the schedule item separated